packages/core/src/shorthands/flex.ts

```diff
@@ -0,0 +1,121 @@
+import * as CSS from 'csstype';
+import type { GriffelStylesStrictCSSObject } from '../types';
+import { FlexInput } from './types';
+
+type FlexStyle = Pick<GriffelStylesStrictCSSObject, 'flexGrow' | 'flexShrink' | 'flexBasis'>;
+
+const isUnit = (value: CSS.Property.Flex | undefined) => typeof value === 'string' && /(\d+(\w+|%))/.test(value);
+
+const isUnitless = (value: CSS.Property.Flex | undefined) => typeof value === 'number' && !Number.isNaN(value);
+
+const isInitial = (value: CSS.Property.Flex | undefined) => value === 'initial';
+
+const isAuto = (value: CSS.Property.Flex | undefined) => value === 'auto';
+
+const isNone = (value: CSS.Property.Flex | undefined) => value === 'none';
+
+const widthReservedKeys = ['content', 'fit-content', 'max-content', 'min-content'] as const;
+type ReservedKeys = typeof widthReservedKeys[number];
+type Width = ReservedKeys | string;
+const isWidth = (value: CSS.Property.Flex | undefined) => widthReservedKeys.some(key => value === key) || isUnit(value);
+
+/**
+ * A function that implements CSS spec conformant expansion for "flex".
+ *
+ * @example
+ *   flex('auto')
+ *   flex(1, '2.5rem')
+ *   flex(0, 0, 'auto')
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/CSS/flex
+ */
+export function flex(...values: FlexInput): FlexStyle {
+  const isOneValueSyntax = values.length === 1;
+  const isTwoValueSyntax = values.length === 2;
+  const isThreeValueSyntax = values.length === 3;
+
+  // One Value Syntax
+  if (isOneValueSyntax) {
+    const [firstValue] = values;
+
+    if (!Array.isArray(firstValue)) {
```

Comment:
```suggestion
```

It cannot be an array, right?

Reply:
That's the fallbacks leftovers, my bad! Thanks for catching 👌